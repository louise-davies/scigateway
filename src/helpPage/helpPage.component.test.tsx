import React from 'react';
import {
  UnconnectedHelpPage,
  CombinedHelpPageProps,
  TableOfContents,
} from './helpPage.component';
import { shallow } from 'enzyme';

jest.mock('../hooks/useAnchor', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('Help page component', () => {
  let props: CombinedHelpPageProps;

  beforeEach(() => {
    props = {
      res: undefined,
    };
  });

  it('should render correctly', () => {
    const wrapper = shallow(<UnconnectedHelpPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render TOC correctly with nested headers', () => {
    props.res = {
      contents:
        "<h2 id='test nested toc item 1'>Nested TOC item</h2>Lorem ipsum dolor sit amet<h3 id='test nested toc item 2'>Nested TOC item 2</h3>consectetur adipiscing elit<br><h2 id='test nested toc item 3'>Nested TOC item 3</h2>Proin suscipit sed nisi ac consectetur",
    };
    const wrapper = shallow(<TableOfContents {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should back to top elements next to each header', () => {
    props.res = {
      contents:
        "<h2 id='test nested toc item 1'>Nested TOC item</h2>Lorem ipsum dolor sit amet<h3 id='test nested toc item 2'>Nested TOC item 2</h3>consectetur adipiscing elit<br><h2 id='test nested toc item 3'>Nested TOC item 3</h2>Proin suscipit sed nisi ac consectetur",
    };
    const wrapper = shallow(<UnconnectedHelpPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
