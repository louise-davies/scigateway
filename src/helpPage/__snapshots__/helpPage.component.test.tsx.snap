// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help page component should back to top elements next to each header 1`] = `
<Styled(div)>
  <ForwardRef(Typography)
    sx={
      Object {
        "color": "secondary.main",
        "fontWeight": "bold",
      }
    }
    variant="h3"
  >
    title
  </ForwardRef(Typography)>
  <TableOfContents
    res={
      Object {
        "contents": "<h2 id='test nested toc item 1'>Nested TOC item</h2>Lorem ipsum dolor sit amet<h3 id='test nested toc item 2'>Nested TOC item 2</h3>consectetur adipiscing elit<br><h2 id='test nested toc item 3'>Nested TOC item 3</h2>Proin suscipit sed nisi ac consectetur",
      }
    }
  />
  <Styled(div)>
    <Styled(ForwardRef(Typography))
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h2 id=\\"test nested toc item 1\\">top-of-page-iconNested TOC item</h2>Lorem ipsum dolor sit amet<h3 id=\\"test nested toc item 2\\">top-of-page-iconNested TOC item 2</h3>consectetur adipiscing elit<br><h2 id=\\"test nested toc item 3\\">top-of-page-iconNested TOC item 3</h2>Proin suscipit sed nisi ac consectetur",
        }
      }
      variant="body1"
    />
  </Styled(div)>
</Styled(div)>
`;

exports[`Help page component should render TOC correctly with nested headers 1`] = `
<Styled(nav)>
  <ForwardRef(Typography)
    variant="h5"
  >
    table-of-contents
  </ForwardRef(Typography)>
  <ul
    dangerouslySetInnerHTML={
      Object {
        "__html": "<li style='list-style: none'><a href='#test nested toc item 1'>Nested TOC item</a></li><ul><li style='list-style: none'><a href='#test nested toc item 2'>Nested TOC item 2</a></li></ul><li style='list-style: none'><a href='#test nested toc item 3'>Nested TOC item 3</a></li>",
      }
    }
    style={
      Object {
        "margin": "0px",
        "padding": "0px",
      }
    }
  />
</Styled(nav)>
`;

exports[`Help page component should render correctly 1`] = `
<Styled(div)>
  <ForwardRef(Typography)
    sx={
      Object {
        "color": "secondary.main",
        "fontWeight": "bold",
      }
    }
    variant="h3"
  >
    title
  </ForwardRef(Typography)>
  <TableOfContents />
  <Styled(div)>
    <Styled(ForwardRef(Typography))
      dangerouslySetInnerHTML={
        Object {
          "__html": "contents",
        }
      }
      variant="body1"
    />
  </Styled(div)>
</Styled(div)>
`;
